yaml/private/lex.nim

```diff
@@ -187,7 +190,10 @@ proc at(lex: YamlLexer, t: typedesc[BaseLexer], pos: int): char {.inline.} =
   lex.blSource.buf[pos]
 
 proc at(lex: YamlLexer, t: typedesc[StringSource], pos: int): char {.inline.} =
-  lex.sSource.src[pos]
+  if lex.sSource.src.len == pos:
+    '\0'
+  else:
+    lex.sSource.src[pos]
```

Comment:
Well that's one way to fix it. However, it makes these checks necessary everywhere. A far simpler (and more performant) solution would be to append `'\0'` to the string when creating a `StringSource`.

Reply:
Ahh, thanks. That's what I would have done, if I understood where I'd have to do that, hehe. I'll check that.